homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,278 @@
+        )
+        self._attr_device_info = DeviceInfo(
+            {
+                "identifiers": {(DOMAIN, str(device.id))},
+                "name": device.label,
+                "manufacturer": MANURFACER_NAME,
+                "model": device_name,
+                "sw_version": "1.0",
+            }
+        )
+
+        parametersDict = {
+            parameter.parameter_code: parameter for parameter in parameters
+        }
+        self.device = device
+        self.available_presets: Parameter | None = parametersDict.get(PARAM_PRESET_MODE)
+        self.available_fan_modes: Parameter | None = parametersDict.get(PARAM_FAN_MODE)
+        self.available_hvac_modes: Parameter | None = parametersDict.get(
+            PARAM_HVAC_MODE
+        )
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        value = self.coordinator.data[self.device.id].state.get_parameter_value(
+            PARAM_CURRENT_TEMPERATURE
+        )
+        if value is None:
+            return None
+        return float(value.value) if value is not None else None
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        value = self.coordinator.data[self.device.id].state.get_parameter_value(
+            PARAM_TARGET_TEMPERATURE
+        )
+        if value is None:
+            return None
+        return float(value.value) if value is not None else None
+
+    @property
+    def preset_modes(self) -> list[str] | None:
+        """Return the available preset modes."""
+        if self.available_presets is None or self.available_presets.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_presets.details
+            if item is not None
+        ]
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_fan_modes.details
+            if item is not None
+        ]
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        preset_mode = self.coordinator.data[self.device.id].state.get_parameter_value(
+            PARAM_PRESET_MODE
+        )
+
+        if (
+            preset_mode is None
+            or self.available_presets is None
+            or self.available_presets.details is None
+        ):
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_presets.details
+                if item is not None and item.state == preset_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            return None
+        return str(val.description)
```

Comment:
```suggestion
        return val.description
```
isn't description already a string?